includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -1584,7 +1629,16 @@ public function is_express_checkout_enabled() {
 	 * @return boolean
 	 */
 	public function is_payment_request_enabled() {
-		return $this->gateway->is_payment_request_enabled();
+		$is_enabled = $this->gateway->is_payment_request_enabled();
+		if ( is_product() ) {
+			return $is_enabled && $this->is_enabled_for_location( 'payment_request', 'product' );
+		} elseif ( $this->is_cart() ) {
+			return $is_enabled && $this->is_enabled_for_location( 'payment_request', 'cart' );
+		} elseif ( $this->is_checkout() ) {
+			return $is_enabled && $this->is_enabled_for_location( 'payment_request', 'checkout' );
+		}
```

Comment:
Nit: this could simply be a sequence of `if` blocks due to the early returns.
```suggestion
		}

		if ( $this->is_cart() ) {
			return $is_enabled && $this->is_enabled_for_location( 'payment_request', 'cart' );
		}

		if ( $this->is_checkout() ) {
			return $is_enabled && $this->is_enabled_for_location( 'payment_request', 'checkout' );
		}
```

Reply:
Agreed ✅ f8b35a6d15d495ade2ac79e58b83216e1d1b8edf